presto-main/src/test/java/com/facebook/presto/cost/TestHistogramCalculator.java

```diff
@@ -0,0 +1,157 @@
+        assertRangeOverlap(Estimate.of(0.1), range(9, 15), uniformHist(range(0, 10)));
+        assertRangeOverlap(Estimate.of(0.1), range(9, 14), uniformHist(range(0, 10)));
+
+        // single edge within, other edge on outer limit
+        assertRangeOverlap(Estimate.of(0.5), range(5, 10), uniformHist(range(0, 10)));
+        assertRangeOverlap(Estimate.of(0.5), range(0, 5), uniformHist(range(0, 10)));
+
+        // infinite ranges
+        assertRangeOverlap(Estimate.of(1.0), range(NEGATIVE_INFINITY, POSITIVE_INFINITY), uniformHist(range(0, 10)));
+        assertRangeOverlap(Estimate.of(0.5), range(5, POSITIVE_INFINITY), uniformHist(range(0, 10)));
+        assertRangeOverlap(Estimate.of(0.5), range(NEGATIVE_INFINITY, 5), uniformHist(range(0, 10)));
+
+        // single value -- since it's not a range, the overlap should be 0%
+        assertRangeOverlap(Estimate.of(0.0), range(5, 5), uniformHist(range(0, 10)));
+        assertRangeOverlap(Estimate.of(0.0), range(0, 0), uniformHist(range(0, 10)));
+        assertRangeOverlap(Estimate.of(0.0), range(10, 10), uniformHist(range(0, 10)));
+
+        // "common" case
+        assertRangeOverlap(Estimate.of(0.3), range(2, 5), uniformHist(range(0, 10)));
+
+        // cases where histogram has unbounded range
+        // -inf
+        assertRangeOverlap(Estimate.unknown(), range(0, 10), uniformHist(range(NEGATIVE_INFINITY, 10)));
+        assertRangeOverlap(Estimate.unknown(), range(9, 10), uniformHist(range(NEGATIVE_INFINITY, 10)));
+        assertRangeOverlap(Estimate.of(0.0), range(10, 10), uniformHist(range(NEGATIVE_INFINITY, 10)));
+        assertRangeOverlap(Estimate.of(0.0), range(10, 12), uniformHist(range(NEGATIVE_INFINITY, 10)));
+        assertRangeOverlap(Estimate.of(0.0), range(11, 12), uniformHist(range(NEGATIVE_INFINITY, 10)));
+        // +inf
+        assertRangeOverlap(Estimate.unknown(), range(0, 10), uniformHist(range(0, POSITIVE_INFINITY)));
+        assertRangeOverlap(Estimate.unknown(), range(0, 1), uniformHist(range(0, POSITIVE_INFINITY)));
+        assertRangeOverlap(Estimate.of(0.0), range(-10, 0), uniformHist(range(0, POSITIVE_INFINITY)));
+        assertRangeOverlap(Estimate.of(0.0), range(-12, -10), uniformHist(range(0, POSITIVE_INFINITY)));
+        assertRangeOverlap(Estimate.of(0.0), range(-12, -11), uniformHist(range(0, POSITIVE_INFINITY)));
+    }
+
+    private static StatisticRange range(double low, double high, double distinctValues)
+    {
+        return new StatisticRange(low, high, distinctValues);
+    }
+
+    private static StatisticRange range(double low, double high)
+    {
+        return new StatisticRange(low, high, NaN);
+    }
+
+    private static StatisticRange unboundedRange(double distinctValues)
+    {
+        return new StatisticRange(NEGATIVE_INFINITY, POSITIVE_INFINITY, distinctValues);
+    }
+
+    private static void assertFilterFactor(StatisticRange range, ConnectorHistogram histogram, Estimate expected)
```

Comment:
nit: Keep the call list order same as below : `Estimate expected, StatisticRange range, ConnectorHistogram histogram)`